guest-agent/VfioService/VfioService/Program.cs

```diff
@@ -18,10 +18,14 @@ static void Main()
             using (var manager = new ClientManager(form))
             {
                 form.ClientManager = manager;
+                InterceptMouse.ClientManager = manager;
+                InterceptMouse.Start();
                 manager.SendCommand(CommandOut.ReportBoot);
 
                 var _ = form.Handle; // create form without showing
+                
                 Application.Run();
+                InterceptMouse.Stop();
```

Comment:
You're right that terminating an application from the Tasks tab sends WM_CLOSE. However:

* This is the parameterless version of `Application.Run()` that is not tied to any specific window and specifically designed to loop forever (this is a hack - the other version would show the window).
* I don't think this can happen anyways as we shouldn't even appear in the Tasks tab since our window is not visible.